cmake/kokkos_arch.cmake

```diff
@@ -412,6 +413,24 @@ if(KOKKOS_ARCH_ZEN3)
   set(KOKKOS_ARCH_AVX2 ON)
 endif()
 
+if(KOKKOS_ARCH_ZEN4)
+  compiler_specific_flags(
+    COMPILER_ID
+    KOKKOS_CXX_HOST_COMPILER_ID
+    Intel
+    -mavx2
+    MSVC
+    /arch:AVX2
+    NVHPC
+    -tp=zen4
+    DEFAULT
+    -march=znver4
+    -mtune=znver4
+  )
+  set(KOKKOS_ARCH_AMD_ZEN4 ON)
+  set(KOKKOS_ARCH_AVX2 ON)
```

Comment:
```suggestion
  set(KOKKOS_ARCH_AVX512 ON)
```

Reply:
That would have no effect.  There is no such macro.

Reply:
We are using `AVX512XEON` pretty much for all architectures and that's the guard used for `SIMD`.

Reply:
Sure, the variable being set in your suggestion is still wrong as far as I can tell (I am on the phone though)

Reply:
> Sure, the variable being set in your suggestion is still wrong as far as I can tell (I am on the phone though)

Updated.

Reply:
It does feel like we should perhaps consolidate or refactor AVX512 macros. There are currently 
 `KOKKOS_ARCH_AVX512MIC` and `KOKKOS_ARCH_AVX512XEON`, but `KOKKOS_ARCH_AVX512MIC` is currently not in use anywhere.

Reply:
I agree and just rename all of them to `KOKKOS_IMPL_AVX512` or so.

Reply:
Let's do the renaming as a follow up